### DISCUSSION TITLE
Setting trace ID and parent observation ID with Python decorator SDK

### DISCUSSION THREAD

**Discussion by MrOrz** [langfuse/langfuse#3186] [Ideas]
### Describe the feature or potential improvement

I love the simplicity of [Python Decorator SDK](https://langfuse.com/docs/sdk/python/decorators) and would like to incorporate distributed tracing with it.

In our setup, we have *LLM apps* that would call *Tools*, with both types of microservices linked to the same Langfuse project. LLM apps would pass the current Langfuse trace ID and observation ID to Tools when the LLM app decide to call tools.

Currently, we are able to set custom trace ID by passing `langfuse_observation_id`  kwarg to the first method that is wrapped by `@observe` in Tools. However, we can't do the same as we also want to set `parent_observation_id`. Although it is [possible](https://langfuse.com/docs/sdk/python/low-level-sdk#span) to set it through the low-level SDK, seems that if we choose to use decorators we just can't achieve the same. 

What is the recommended way to set parent observation ID through the decorator based Python SDK? A documentation / snippet on the recommended way to do Distributed Tracing between the Tool and LLM apps would be helpful too.

### Additional information

Original Discord thread: https://discord.langfuse.com/m/1278314469960974379

Other related discussions
- Maintainer suggests a private API that tinkers with the root trace ID https://github.com/orgs/langfuse/discussions/2769#discussioncomment-10184014
- Related conversation without follow-ups https://discord.langfuse.com/m/1259866301879287959

**Comment by marcklingen** [comment]
@MrOrz, agree that an "official" solution to distributed tracing when using the decorator-based integration is helpful here. I'd recommend to use the rootTraceId-based workaround suggested by @hassiebp in the meantime. Does it work for you?

**Comment by MrOrz** [comment]
I tried using `langfuse_context._set_root_trace_id` to set trace ID, but somehow the trace ID is not set. I can try providing a reproduction.

Setting trace ID via `langfuse_observation_id` do work, although it requires an additional wrapper outside `@observe()`-ed method to provide the kwarg..

**Reply by marcklingen** [reply]
Got it, that's not ideal. I created this issue to track the implementation of this: https://github.com/langfuse/langfuse/issues/3190

**Comment by hassiebp** [comment]
@MrOrz We have just released support for distributed tracing via passing a Langfuse parent ID. Please check out our release and docs and curious to hear whether all works for you 👍🏾 

- [Docs](https://langfuse.com/docs/sdk/python/decorators#set-parent-trace-id-or-parent-span-id)
- [Release](https://github.com/langfuse/langfuse-python/releases/tag/v2.50.0)
